module-2/backend/week_one.md

```diff
@@ -5,28 +5,71 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+- Sinatra is a framework; a DSL language that allows us to manipulate code by eliminating extra work that’s automatically taken care of in the background. 
+
+3. What is MVC?
+- Model, View, Control: These are sub folders that live within our app parent folder. A framework that allows you to build apps in a manner that is convention.
+
+4. Why do we follow conventions when creating our actions/path names in our Sinatra routes?
+-There is a order of precedence, plus most programmers are used to using a specific form of programming. 
+
+5. What types of variables are accessible in our view templates without explicitly passing them?
+-instance variables
+
+6. Given the following block of code, how would I pass an instance variable count with a value of 1 to my index.erb template?
+get '/horses' do
+  erb :index
+end
+- @count = 1
+
+7. In the same code block, how would I pass a local variable name with a value of Mr. Ed to the view?
+- name = ‘Mr. Ed’ 
+- :locals => {name: name}
+
+8. What's the purpose of ERB?
+- Embedded Ruby (ERB), allows you to add ruby code into html files
+
+9. Why do I need a development AND test database?
+- You need a development and test database to be able to build your app without effecting your production database. Having both also allows you to use similar data to help build your app.
+
+10. What is CRUD and why is it important?
+- Create, Read, Update, Destroy: these are the basic functions that guide and help build our app.
+
+11. What does HTTP stand for?
+- Hyper Text Transfer Protocol
+
+12. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+- <%= %> and <% %>, one “prints” what you wanna see from a user view, the other doesn’t output
+
+13. What's an ORM?
+- Object Relational Mapping- allows you to manipulate objects between databases.
+
+14. What's the most commonly used ORM in ruby (Sinatra & Rails)?
+- Active Record
+
+15. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+- GET/restaurants  — takes you to the index/dashboard
+- GET/restaurants/new — uses the HTML form to create new instance  
+- POST/restaurants —  create a new restaurant instance 
+- GET/restaurants/id — shows an instance of a restaurant 
+- GET/restaurants/id/edit — uses HTML form to edit instance
+- DELETE/restaurants/id — deletes an instance using HTML form
+
+16. What's a migration?
+- Allows you to create a schema with ruby
+
+17. When you create a migration, does it automatically modify your database?
+- Unless you run “rake db:migrate” after you build the instructions for how you want to schema to look, then no.
+
+18. How does a model relate to a database?
+- The model uses the database to pull data and with it adds functionality to your app.

```

Comment:
Yup! I would also say that a model corresponds to a table in the database, while an instance of that model corresponds to a particular row in that table.